cmd/kusk/cmd/deploy.go

```diff
@@ -79,14 +80,20 @@ var deployCmd = &cobra.Command{
 	SilenceErrors: true,
 	SilenceUsage:  true,
 	Long:          ``,
+	PreRunE: func(cmd *cobra.Command, args []string) error {
+		if file != "" && overlaySpecPath != "" {
+			return fmt.Errorf(`'-i, --in and --overlay are mutually exclusive`)
+		}
+		return nil
+	},
```

Comment:
This is repeated in generate
This is why i feel these commands grouped under an `api` parent command that can handle this logic

Reply:
I couldn't agree more on that. It was only natural to do this in PreRun on the parent cmd